### DISCUSSION TITLE
improve shadow quality with sparse vsm

### DISCUSSION THREAD

**Discussion by octanejohn** [godotengine/godot-proposals#9021] [3D]
this idea should be for near field
https://ktstephano.github.io/rendering/stratusgfx/svsm

for far field sdf shadows could be used but are too limiting

another idea i am trying to see if it works with far shadows is this(maybe works with other ideas too? gi, ao, far reflection, gpu occlution culling?)
https://zheng95z.github.io/publications/roma23

**Comment by Calinou** [comment]
Note that implementing https://github.com/godotengine/godot-proposals/issues/3908 should improve directional shadow quality a fair bit already, without going through the added complexity of virtual shadow maps.
